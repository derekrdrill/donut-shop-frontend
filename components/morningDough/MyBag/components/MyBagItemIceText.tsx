import * as React from 'react';
import { Typography } from '@mui/material';

import { MyBagItem } from '../../../../context/types/MyBagItem';

interface MyBagItemIceTextProps {
  myBagItem: MyBagItem;
}

const MyBagItemIceText = ({ myBagItem }: MyBagItemIceTextProps) =>
  myBagItem.subCategory === 'coldDrinks' && !myBagItem.bottled && myBagItem.ice ? (
    <Typography data-testid='ice-text' component='p' variant='caption'>
      {`${myBagItem.ice.toUpperCase()} ICE`}
    </Typography>
  ) : (
    <React.Fragment />
  );

export default MyBagItemIceText;
