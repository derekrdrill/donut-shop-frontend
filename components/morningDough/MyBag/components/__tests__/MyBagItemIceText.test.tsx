import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import MyBagItemIceText from '../MyBagItemIceText';

describe('MyBagItemIceText unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(
        <MyBagItemIceText
          myBagItem={{
            orderID: '1',
            menuItemID: 'icedCoffee',
            name: 'Iced Coffee',
            category: 'drink',
            subCategory: 'coldDrinks',
            quantity: 3,
            dairy: null,
            flavor: null,
            sugar: false,
            size: 'md',
            ice: null,
            creamCheese: null,
            butter: null,
            bottled: false,
            soda: false,
            menuItemImage: 'icedCoffee.png',
          }}
        />,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders MyBagItemIceText correctly (ice-text)', async () => {
    render(
      <MyBagItemIceText
        myBagItem={{
          orderID: '1',
          menuItemID: 'icedCoffee',
          name: 'Iced Coffee',
          category: 'drink',
          subCategory: 'coldDrinks',
          quantity: 3,
          dairy: 'halfAndHalf',
          flavor: 'frenchVanilla',
          sugar: false,
          size: 'md',
          ice: 'light',
          creamCheese: null,
          butter: null,
          bottled: false,
          soda: false,
          menuItemImage: 'icedCoffee.png',
        }}
      />,
    );

    const menuItemIceText = await screen.findAllByTestId('ice-text');

    expect(menuItemIceText[0]).toHaveClass('MuiTypography-root');
    expect(menuItemIceText[0]).toHaveTextContent('LIGHT ICE');
  });
});
