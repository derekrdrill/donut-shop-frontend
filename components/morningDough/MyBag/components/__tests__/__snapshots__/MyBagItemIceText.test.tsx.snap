// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBagItemIceText unit tests renders as expected 1`] = `null`;
